pkg/server/endpoints/registration/handler.go

```diff
@@ -285,6 +287,36 @@ func (h *Handler) ListBySpiffeID(ctx context.Context, request *registration.Spif
 	}, nil
 }
 
+//ListAllEntriesWithPages retrieves all registered entries with pagination.
+func (h *Handler) ListAllEntriesWithPages(ctx context.Context, request *registration.ListAllEntriesRequest) (_ *registration.ListAllEntriesResponse, err error) {
+	counter := telemetry_registrationapi.StartListEntriesCall(h.Metrics)
+	telemetry_common.AddCallerID(counter, getCallerID(ctx))
+	defer counter.Done(&err)
+	log := h.Log.WithField(telemetry.Method, telemetry.ListAllEntriesWithPages)
+
+	ds := h.getDataStore()
+	if request.PageSize == 0 {
+		request.PageSize = _defaultListEntriesPageSize
+	}
+	fetchResponse, err := ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{
```

Comment:
This isn't something we need to address in this PR, but eventually I think the datastore will need to ability to reduce the page size dynamically when the result will be too large to fit in a single message.

Reply:
Agreed. This would be cool to have.